### PR TITLE
add TimeoutException base class for all timeout exceptions

### PR DESCRIPTION
**Pull request by oczkers**
It would be nice to have one exception object to catch on higher level in httpx but implementing it here and importing would be probably cleanest way to do it.

httpx had Timeout**Exception** so You might want to keep it for backward compatibility.

### CODE REVIEW

**Comment by tomchristie** [comment]
Yup probably makes sense.
We'd want to expose it in the `__init__` too.

**Comment by oczkers** [comment]
Looks like TimeoutError namespace is [reserved in python3](https://docs.python.org/3.5/library/exceptions.html#TimeoutError) so i've renamed it to TimetoutException and included import in __init__

**Comment by tomchristie** [comment]
I guess an alternative here would be us subclassing Python's built-in `TimeoutError`.

Any thoughts @encode/maintainers ?

**Comment by florimondmanca** [comment]
+1 from me on a common timeout exception class.

Not sure if using the builtin as a base is a good idea or not, though. Eg Requests and others have their own custom class (although awkwardly named after the builtin), and all our other exceptions are custom...

So I’d go for a custom base `TimeoutException`.

File: httpcore/__init__.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     ProxyError,
     ReadError,
     ReadTimeout,
+    TimeoutException,
```
Comment (florimondmanca) [inline httpcore/__init__.py:14]: Would need to add it to `__all__` too. :)

**Comment by oczkers** [comment]
> g Requests and others have their own custom class (although awkwardly named after the builtin), and all our other exceptions are custom...

It might be crazy but how about custom base class that inherits on builtin?
```python
class TimeoutException(TimeoutError):
```

File: httpcore/__init__.py (hunk 1 above)
Comment (oczkers) [reply httpcore/__init__.py:14]: done, sorry :smile:

**Pull request merged**